backend/controllers/users.js

```diff
@@ -115,27 +115,33 @@ const editPost = async (req,res,next)=>{
         cover:filename
         
     })
-        res.json(postDoc)
+        res.status(200).json(postDoc)
     } catch (error) {
         next(error)
     }
 }
 
 const deletePost =async(req,res,next)=>{
 
-    const {id} = req.body
+    const {id} = req.params
     const {token} = req.cookies
     try {
 
         // const post = await Post.findByIdAndDelete(id)
-        const authorId = await Post.findById(id)
+        const postDoc = await Post.findById(id)
+        if(!postDoc){
+            return next(new BadRequest("Post Not Found"))
+        
+        }
         const data = await jwt.verify(token,process.env.JWT_SECRET)
```

Comment:
Could you refactor this part into a middleware, this part is repeated in multiple routes.
You could do something like this.
```js
const authenticatedMiddleware = (req, res, next) => {
    const {token} = req.cookies;
    
    // Make sure that *token* exists and return Unauthenticated if not

    const data = await jwt.verify(token, process.env.JWT_SECRET);

   // Same thing here but for data.
   // You might also need to put the jwt.verify code in a try/catch block in case somebody modified their token
   // which will make it throw an error

   // Finally you can associate the jwt payload into the request so you can use it later in the routes
   req.user = data;

   // Then in the deletePost route for example you can do 
   const isAuthor = postDoc.author.equals(req.user.id);
};
```